lib/internal/process/per_thread.js

```diff
@@ -295,15 +298,16 @@ function buildAllowedFlags() {
   const nodeFlags = new SafeSet(ArrayPrototypeMap(allowedNodeEnvironmentFlags,
                                                   trimLeadingDashes));
 
-  class NodeEnvironmentFlagsSet extends SafeSet {
-    constructor(...args) {
-      super(...args);
+  class NodeEnvironmentFlagsSet extends Set {
+    #set;
+    constructor(iterable) {
+      super();
+      this.#set = new SafeSet(iterable);
```

Comment:
While this likely is not a very performance sensitive path, using a private property here still caries a significant performance penalty. Should consider using a hidden symbol instead